### PR TITLE
fix: bug where player 2 is immortal even after hp is below 0

### PR DESCRIPTION
**Pull request by Failzuma**

### CODE REVIEW

File: .idea/misc.xml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_22" default="true" project-jdk-name="22" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_19" default="true" project-jdk-name="22" project-jdk-type="JavaSDK">
```
Comment (SEOKMINJUN) [inline .idea/misc.xml:3]: Please revert jdk version to 22.
Then i will accept it.

File: .idea/workspace.xml
Comment (SEOKMINJUN) [inline .idea/workspace.xml:1]: I don't care workspace file much, 
but please don't commit including this file next time.

**Review by SEOKMINJUN — COMMENTED** [review]
Please revert file in .idea folder than i approve it.

File: .idea/misc.xml (hunk 1 above)
Comment (Failzuma) [reply .idea/misc.xml:3]: ah my bad, i forgot about that one

File: .idea/workspace.xml
Comment (Failzuma) [reply .idea/workspace.xml:1]: might be good if its included in .gitignore

**Pull request merged**